pkg/streamingpromql/optimize/ast/propagate_matchers.go

```diff
@@ -162,31 +163,52 @@ func updateVectorSelectorWithAggregationInfo(vs *enrichedVectorSelector, groupin
 	vs.containsAgg = true
 }
 
-// TODO: Consider more functions, and add tests for these.
-func vectorSelectorArgumentIndex(funcName string) int {
+func VectorSelectorArgumentIndex(funcName string) (int, error) {
 	// If the function is eligible for matcher propagation, return the index of the argument
 	// that contains the vector selector (may be embedded in other kinds of expressions).
 	// Otherwise, return -1.
 	switch funcName {
+	// Simple
+	case "absent", "changes", "resets":
+		return 0, nil
```

Comment:
Could you please add some tests specifically for `absent` and `absent_over_time`? 

I suspect we'll have to exclude them from this optimisation, as adding labels to the selector inside an `absent`/`absent_over_time` call changes the labels returned by the function.

Reply:
Good catch, thanks!